src/components/application_manager/test/application_impl_test.cc

```diff
@@ -740,16 +741,37 @@ TEST_F(ApplicationImplTest, SuspendNaviStreaming) {
   app_impl->SuspendStreaming(type);
 }
 
-TEST_F(ApplicationImplTest, SuspendAudioStreaming) {
+TEST_F(ApplicationImplTest, SuspendAudioStreaming_StreamingApproved) {
+  app_impl->set_audio_streaming_approved(true);
   protocol_handler::ServiceType type = protocol_handler::ServiceType::kAudio;
   EXPECT_CALL(mock_application_manager_, OnAppStreaming(app_id, type, false));
   EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
               SendOnDataStreaming(type, false, _));
   app_impl->SuspendStreaming(type);
 }
 
+TEST_F(ApplicationImplTest, SuspendNaviStreaming_StreamingNotApproved) {
+  app_impl->set_video_streaming_approved(false);
+  protocol_handler::ServiceType type =
+      protocol_handler::ServiceType::kMobileNav;
+  EXPECT_CALL(mock_application_manager_, OnAppStreaming(app_id, type, false));
+  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
+              SendOnDataStreaming(type, false, _)).Times(0);
+  app_impl->SuspendStreaming(type);
+}
+
+TEST_F(ApplicationImplTest, SuspendAudioStreaming_StreamingNotApproved) {
+  app_impl->set_audio_streaming_approved(false);
+  protocol_handler::ServiceType type = protocol_handler::ServiceType::kAudio;
+  EXPECT_CALL(mock_application_manager_, OnAppStreaming(app_id, type, false));
+  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
+              SendOnDataStreaming(type, false, _)).Times(0);
+  app_impl->SuspendStreaming(type);
+}
+
 // TODO {AKozoriz} : Fix tests with streaming (APPLINK-19289)
-TEST_F(ApplicationImplTest, DISABLED_Suspend_WakeUpAudioStreaming) {
+TEST_F(ApplicationImplTest, Suspend_WakeUpAudioStreaming_StreamingApproved) {
```

Comment:
Please disable these tests, they fail for an unrelated reason as mentioned before:
1. `Suspend_WakeUpAudioStreaming_StreamingApproved`
2. `Suspend_WakeUpNaviStreaming_StreamingApproved`

The issue with these is that they start a timer in the `WakeUpStreaming` function that fires after the test is done, causing a crash.

Another option might be to modify this by adding another corresponding `SuspendStreaming` so that this timer is cancelled before the end of the test.